tests/annotation/common/src/main/java/sparklr/common/AbstractRefreshTokenSupportTests.java

```diff
@@ -32,10 +32,36 @@ public void testHappyDay() throws Exception {
 
 		// now use the refresh token to get a new access token.
 		assertNotNull(accessToken.getRefreshToken());
-		OAuth2AccessToken newAccessToken = refreshAccessToken(accessToken.getRefreshToken().getValue());
+		OAuth2AccessToken newAccessToken = refreshAccessToken(accessToken.getRefreshToken().getValue(),
+			"my-trusted-client", "read");
 		assertFalse(newAccessToken.getValue().equals(accessToken.getValue()));
 
 		verifyAccessTokens(accessToken, newAccessToken);
+
+		cancelToken(accessToken.getValue());
+		cancelToken(newAccessToken.getValue());
+
+	}
+
+	/**
+	 * Tests that refreshing an access token specifying no scopes for a client
+	 * with unlimited scopes returns an access token with the originally
+	 * granted scopes.
+	 */
+	@Test
+	public void testGrantedScope() throws Exception {
+
+		OAuth2AccessToken accessToken = getAccessToken("read write", "my-client-with-empty-scopes");
```

Comment:
This call should fail given that `my-client-with-empty-scopes` is configured with `null` scopes and the request contains `requested` scopes. The Authorization Server should only grant scopes if the `requested` scopes **are a subset** of the `registered` scopes.

Reply:
I'm not sure this is right -- from the [docs](https://projects.spring.io/spring-security-oauth/docs/oauth2.html#configuring-client-details):

> `scope`: The scope to which the client is limited. If scope is undefined or empty (the default) the client is not limited by scope.

Thus a client with empty scopes accepts **any** requested scopes.